commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-test-tests/src/test/java/dev/jorel/commandapi/test/ConfigGenerationTests.java

```diff
@@ -0,0 +1,208 @@
+		), adapter);
+	}
+
+	@Test
+	public void testConfigOptionAddition() {
+		CommentedConfigOption<Boolean> createDispatcherJson = new CommentedConfigOption<>(new String[] {
+			"Create dispatcher JSON (default: false)",
+			"If \"true\", the CommandAPI creates a command_registration.json file showing the",
+			"mapping of registered commands. This is designed to be used by developers -",
+			"setting this to \"false\" will improve command registration performance."
+		}, false);
+
+		bukkitConfig.getAllOptions().put("create-dispatcher-json", createDispatcherJson);
+		generator = ConfigGenerator.createNew(bukkitConfig);
+		BukkitConfigurationAdapter updatedAdapter = (BukkitConfigurationAdapter) generator.generate(adapter);
+
+		validateConfigOptions(Set.of(
+			"silent-logs", "verbose-outputs", "messages.missing-executor-implementation", "create-dispatcher-json"
+		), updatedAdapter);
+
+		validateConfigOptionComments(Map.ofEntries(
+			Map.entry("silent-logs", silentLogs.comment()),
+			Map.entry("verbose-outputs", verboseOutputs.comment()),
+			Map.entry("messages.missing-executor-implementation", missingExecutorImplementation.comment()),
+			Map.entry("create-dispatcher-json", createDispatcherJson.comment()),
+			Map.entry("messages", messages.comment())
+		), updatedAdapter);
+	}
+
+	@Test
+	public void testConfigOptionDeletion() {
+		bukkitConfig.getAllOptions().remove("silent-logs");
+		generator = ConfigGenerator.createNew(bukkitConfig);
+		BukkitConfigurationAdapter updatedAdapter = (BukkitConfigurationAdapter) generator.generate(adapter);
+
+		validateConfigOptionsAbsent(Set.of("silent-logs"), updatedAdapter);
+	}
+
+	@Test
+	public void testConfigOptionCommentUpdate() {
+		silentLogs = new CommentedConfigOption<>(new String[] {
+			"Defines if silent logs should happen"
+		}, false);
+
+		bukkitConfig.getAllOptions().put("silent-logs", silentLogs);
+		generator = ConfigGenerator.createNew(bukkitConfig);
+		BukkitConfigurationAdapter updatedAdapter = (BukkitConfigurationAdapter) generator.generate(adapter);
+
+		validateConfigOptionComments(Map.ofEntries(
+			Map.entry("silent-logs", silentLogs.comment())
+		), updatedAdapter);
+	}
+
+	@Test
+	public void testNestedSections() {
+		CommentedConfigOption<Boolean> subSubOption = new CommentedConfigOption<>(new String[0], false);
+
+		bukkitConfig.getAllOptions().put("root.nested.option", subSubOption);
+		generator = ConfigGenerator.createNew(bukkitConfig);
+		BukkitConfigurationAdapter updatedAdapter = (BukkitConfigurationAdapter) generator.generate(adapter);
+
+		validateSections(List.of("root", "nested"), "option", updatedAdapter.config());
+	}
+
+	// Test methods
+	public void validateConfigOptions(Set<String> options, BukkitConfigurationAdapter adapter) {
+		boolean containsAll;
+		for (String option : options) {
+			containsAll = adapter.contains(option);
+			if (!containsAll) {
+				throw new IllegalStateException("Config option '" + option + "' does not exist!");
+			}
+		}
```

Comment:
Since we're in the test environment, using the methods from `org.junit.jupiter.api.Assertions` here and in the other `validate...` methods would match the other tests. I think the `AssertionFailedError`s thrown by `org.junit.jupiter.api.Assertions` also make it clearer if the tests fail that it was an unexpected result in the tests, and not a general `IllegalStateException` thrown by the production code.

```suggestion
		for (String option : options) {
			assertTrue(adapter.contains(option), "Config option '" + option + "' does not exist!");
		}
```